src/contraction/src/pgr_contract.hpp

```diff
@@ -0,0 +1,379 @@
+					float dist=graph.graph[edesc].cost;
+					
+					if (min_distance-dist>=0.000001)
+					{
+						min_distance=dist;
+					}
+					else
+					{
+						std::cout << "Updating shortcut " << "(" << eid1 << ", " << eid2 << ")" << " with " << min_distance << endl;
+						graph.graph[edesc].cost=min_distance;
+						graph.graph[edesc].type=1;
+					}
+				}
+				else
+				{
+					shortcut.id=++last_edge_id;
+					eid=shortcut.id;
+					shortcut.source=tid1;
+					shortcut.target=tid2; 
+					shortcut.cost=min_distance;
+					shortcut.type=2;
+					graph.graph_add_edge(shortcut);
+				}
+				psuedoEdges[eid]=make_pair(eid1,eid2);
+				Edge removed_edge;
+				for (int i = 0; i < 2; ++i)
+				{
+					V t=neighbors_desc[i];
+					removed_edge=graph.graph[edge(front, t,graph.graph).first];
+					removedVertices[front_id].push_front(removed_edge);
+				}
+				graph.disconnect_vertex(front_id);
+				graph.m_num_vertices--;
+			}
+		}
+		two_degree_vertices_0.erase(two_degree_vertices_0.begin(),
+			two_degree_vertices_0.begin()+1);
+
+	}
+}*/
+
+template < class G >
+int64_t
+Pgr_contract< G >::getGraph_string(G &graph,std::ostringstream& estring)
+{
+	V_i vi;
+	EO_i out,out_end;
+	int64_t count=0;
+	for (vi = vertices(graph.graph).first; vi != vertices(graph.graph).second; ++vi) {
+		for (boost::tie(out, out_end) = out_edges(*vi, graph.graph);
+			out != out_end; ++out)
+		{
+			V source_desc=source(*out, graph.graph);
+			V target_desc=target(*out, graph.graph);
+			int64_t source_id=graph.graph[source_desc].id;
+			int64_t target_id=graph.graph[target_desc].id;
+
+			if (target_desc>source_desc)
+			{
+				estring << graph.graph[*out].id << ","
+				<< source_id << "," << target_id << ","
+				<< graph.graph[*out].cost << ","
+				<< graph.graph[*out].reverse_cost << "$";
+				count++;
+			}
+
+		}
+	}
+	return count;
+}
+
+template < class G >
+void
+Pgr_contract< G >::getRemovedE_string(G &graph,std::ostringstream& estring)
+{
+	for (removed_E_i iter = graph.removed_edges.begin(); iter != graph.removed_edges.end(); iter++)
+	{
+		// Edge temp=*iter ;
+		estring << iter->id << "," << iter->source << ","
+		<< iter->target << "," << iter->cost << "$"; 
+		// << "," << iter->reverse_cost << "$";
+	}
+}
+
+template < class G >
+void
+Pgr_contract< G >::getRemovedV_string(std::ostringstream& vstring)
+{
+	for (removed_V_i iter = removedVertices.begin(); iter != removedVertices.end(); iter++)
+	{
+		int64_t vid=iter->first;
+		std::deque<Edge>::iterator edge_iter;
+		for (edge_iter = iter->second.begin(); edge_iter != iter->second.end(); edge_iter++)
+		{
+			Edge temp=*edge_iter ;
+			vstring << vid <<","<< temp.id << ","
+			<< temp.source << "," << temp.target << ","
+			<< temp.cost << "," << temp.reverse_cost << "$";
+		}
+		//std::cout << endl;
+	}		
+
+}
+
+template < class G >
+void
+Pgr_contract< G >::getPsuedoE_string(std::ostringstream& pstring)
+{
+	for (psuedo_E_i iter = psuedoEdges.begin(); iter != psuedoEdges.end(); iter++)
```

Comment:
for (const auto pe : pseudoEdges) {
   pstring << pe.fist << ',' << pe.second.first  ...etc